BlueprintUI/Sources/Element/UIViewElement.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  UIViewElement.swift
+//  BlueprintUI
+//
+//  Created by Kyle Van Essen on 6/4/20.
+//
+
+import UIKit
+
+
+///
+/// An element type  which makes it easier to wrap an existing `UIView` instance that
+/// provides its own sizing via `sizeThatFits`. An instance of the view is used for
+/// sizing and measurement, so that you do not need to re-implement your own measurement.
+///
+/// Example
+/// -------
+/// If you were implementing a very basic `Label` element, your implementation would look something
+/// like this:
+/// ```
+/// struct Label : UIViewElement {
+///
+///     var text : String
+///     var font : UIFont
+///
+///     typealias UIViewType = UILabel
+///
+///     static func makeUIView() -> UILabel {
+///         UILabel()
+///     }
+///
+///     func updateUIView(_ view: UILabel) {
+///         view.text = self.text
+///         view.font = self.font
+///     }
+/// }
+/// ```
+public protocol UIViewElement : Element {
+    
+    /// The type of the view associated with the element.
+    associatedtype UIViewType : UIView
+        
+    /// Create and return a new instance of the provided view type.
+    ///
+    /// Note
+    /// ----
+    /// Ensure that you do not pass any values to the initializer of your view type
+    /// that you cannot also update in `updateUIView(_:)`, as view instances
+    /// are reused for sizing and measurement.
+    static func makeUIView() -> UIViewType
+
+    /// Update the view instance with the content from the element.
+    ///
+    /// Example
+    /// -------
+    /// If you were to implement a simple `UIViewElement` which wraps a `UILabel`,
+    /// your update method would look like this:
+    /// 
+    /// ```
+    /// func updateUIView(_ view: UILabel) {
+    ///     view.text = self.text
+    ///     view.font = self.font
+    /// }
+    /// ```
+    func updateUIView(_ view: UIViewType)
+    
+    /// Returns the sizing measurement for the element for the provided
+    /// measurement view.
+    ///
+    /// You usually do not need to implement this method – the default implementation of
+    /// this method simply calls `sizeThatFits(_:)` on the provided view.
+    ///
+    /// The view is fully configured and updated before this method is called – you do not need to
+    /// update it in any way.
+    ///
+    /// When To Override
+    /// ----------------
+    /// You may want to override this method if you need to mutate the value returned from `sizeThatFits(_:)`,
+    /// for example to round it up to an integral pixel value in the case of a `UILabel`, or if you
```

Comment:
I think this is probably a bad example to have in the docs, for two reasons:
a) UILabel does its own rounding, so this isn't a valid use case
b) I don't want to encourage folks to make their own label wrappers

Reply:
Oh right – UILabel rounds, the text measurement stuff doesnt. Always mix those up.

Reply:
Updated this to reference UIImageView instead